internal/monitor/monitor.go

```diff
@@ -46,42 +59,89 @@ func New(logger *zap.Logger, k8s kubernetes.Interface, cfg api.Config) (*Monitor
 	}, nil
 }
 
+func (m *Monitor) getScheduledCommandJobs(
+	ctx context.Context,
+	tag string,
+) (
+	jobs []*api.JobJobTypeCommand,
+	fatal bool,
+	err error,
+) {
+	logger := m.logger
+	if m.cfg.ClusterUUID == "" {
+		jobsResp, err := api.GetScheduledJobs(ctx, m.gql, m.cfg.Org, []string{tag})
+		if err != nil {
+			if !errors.Is(err, context.Canceled) {
+				logger.Warn("failed to retrieve builds for pipeline", zap.Error(err))
+			}
+			return nil, false, err
+		}
+
+		if jobsResp.Organization.Id == nil {
+			return nil, true, fmt.Errorf("invalid organization: %s", m.cfg.Org)
+		}
+
+		jobs = make([]*api.JobJobTypeCommand, 0, len(jobsResp.Organization.Jobs.Edges))
+		for _, job := range jobsResp.Organization.Jobs.Edges {
+			jobs = append(jobs, job.Node.(*api.JobJobTypeCommand))
+		}
+	} else {
```

Comment:
[Optional] Probably worth considering dividing the the method into two sub-methods, one for each of the branches.

Reply:
I think this is more of a symptom of how the generated methods return 2 struct types, `GetScheduledJobsReponse` and `GetScheduledJobsClusteredResponse` that have the same structure and need to be handled identically, but the type system won't let us write only one block of code to handle both.

I decided to rectify this using an interface to abstract over the parts of the structs we need. There's still a bit of duplication, but it is limited to identical implementations of the interface methods only do "one thing". If go had a more expressive syntax, both of these methods could be one-liners. In any case, this should be earlier to maintain that larger duplicated blocks that do multiple things.